compiler/defer.go

```diff
@@ -34,6 +35,40 @@ func (c *Compiler) deferInitFunc(frame *Frame) {
 	c.builder.CreateStore(llvm.ConstPointerNull(deferType), frame.deferPtr)
 }
 
+// isCyclic checks if there is a path from a basic block to itself.
+func isCyclic(start *ssa.BasicBlock) bool {
```

Comment:
I think the name `isCyclic` is not immediately obvious in what it means. What about `isInLoop`?